selfdrive/car/toyota/interface.py

```diff
@@ -110,6 +110,19 @@ def get_params(candidate, fingerprint=gen_empty_fingerprint(), car_fw=[], experi
       ret.steerRatio = 14.3
       tire_stiffness_factor = 0.7933
       ret.mass = 3585. * CV.LB_TO_KG + STD_CARGO_KG  # Average between ICE and Hybrid
+      ret.lateralTuning.init('pid')
+      ret.lateralTuning.pid.kpV = [0.6]
+      ret.lateralTuning.pid.kiV = [0.1]
+      ret.lateralTuning.pid.kf = 0.00007818594
+
+      # 2019+ RAV4 TSS2 uses two different steering racks and specific tuning seems to be necessary.
+      # See https://github.com/commaai/openpilot/pull/21429#issuecomment-873652891
+      for fw in car_fw:
+        if fw.ecu == "eps" and (fw.fwVersion.startswith(b'\x02') or fw.fwVersion in [b'8965B42181\x00\x00\x00\x00\x00\x00']):
```

Comment:
Heads up, the single byte prefix on some FW versions is simply a length, so it may not describe what you want (though I still haven't figured out why some cars have different lengths). For example, new RAV4s not using torque have a length 2 EPS FW, as well as the radar ACC RAV4s. Have we looked into the 2022s to make sure they need this different tuning? @nuwandavek